keras/engine/topology.py

```diff
@@ -465,6 +465,17 @@ def __call__(self, x, mask=None):
             x: can be a tensor or list/tuple of tensors.
             mask: tensor or list/tuple of tensors.
         '''
+        # place the op on the right device:
+        if type(x) in [list, tuple] and not hasattr(x[0], '_keras_on_device'):
```

Comment:
Nothing that I can think of.  There is no global device attribute for theano. Only API is the `target` argument for `shared`  and the `.transfer()` method. So whenever we do something at higher level Keras on a non-default device we have to communicate with the backend.